src/adv.py

```diff
@@ -49,3 +83,103 @@
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+
+def invalid():
+    print(f'Invalid Direction! You are currently {current_player.current_room.name}\n')
+    direction = input(f'Where would you like to go? ')
+
+
+while True:
+    print(f'You are currently in {current_player.current_room.name}\n')
+    print('To pick up or drop an item just enter the name of the item you see!\n')
+    print(f'{current_player.current_room.description}, Items: {", ".join([item.name for item in current_player.current_room.items])}\n')
+    direction = input(f'What now? ')
+    if direction.lower() in valid_directions:
+        try:
+            current_player.move(direction)
+            print(f'{current_player.current_room.name}')
+        except:
+            invalid()
+    elif direction.lower() == 'q':
+        print('Thanks for playing!\n')
+        break
+    elif direction.lower() in valid_items and 'get' or 'drop':
+        if direction.lower().split()[1] == 'compass':
+            try:
```

Comment:
interesting way to implement this. I'm not sure if this would work but maybe a function could do this for you and make it more DRY. where you pass in the string of the item you are trying to pickup and then inside that function you can do the try/except inside of the function. but good job with this project :)